datalad/core/local/status.py

```diff
@@ -495,3 +495,24 @@ def custom_result_summary_renderer(results):  # pragma: more cover
                for r in results):
             from datalad.ui import ui
             ui.message("nothing to save, working tree clean")
+
+        # ### Report on possibly cleanable temporary locations
+
+        # Infer the dataset we ran on from the results and get original
+        # recursive (+limit) options:
+        refds = set(r['refds'] for r in results)
+        assert len(refds) == 1
+        refds = Dataset(refds.pop())
+        recursive = kwargs.pop('recursive', False)
+        recursion_limit = kwargs.pop('recursion_limit', None)
+        clean_infos = refds.clean(info=True,
```

Comment:
I really don't like such an approach:
- "renderer" which is also a "do house-keeping" action, possibly even failing along its operation, is IMHO "too much"
- `status` operation could have been limited to some subdatasets, and how this would go through the entire hierarchy of 1000s I have? thank you not
- in my view `status` should be a read-only operation, and cleanup shouldn't be its side-effect
- despite it might be a nice thing to have for some cases, but clearly not for all (e.g. I might not like to cleanup anything).

Reply:
d'oh -- ok, blind me now mentioned this new `info=True` so operation would be a read-only, crossed out some items ;-).  Still not sure about the last one wondering if it is wise to add more extra run time on top of already not that speedy `status` when operating on heavy hierarchies.

Reply:
> status operation could have been limited to some subdatasets, and now this would go through the entire hierarchy of 1000s I have?

You mean it should consider `path` passed into `status` in addition to `recursion_limit`, @yarikoptic ?

Reply:
Re "extra run time":

Consider responding to https://github.com/datalad/datalad/pull/5738#discussion_r650906698? ;-)

Reply:
> You mean it should consider `path` passed into `status` in addition to `recursion_limit`, @yarikoptic ?

well, it sounds like that -- or may be any of those extra kwargs aren't needed?  may be paths for relevant subdatasets could be taken from the `results` records passed into `status`? then `clean` would be checked on those datasets (without any recursion)

Reply:
>  may be paths for relevant subdatasets could be taken from the results records passed into status? 

True. Will do that. Makes the signature change independent from this PR (still think a unification w/ result renderer makes sense).

Reply:
continuing on "renderer" taking on new tasks, I wondered: wouldn't it be better to have `status` code to actually do the clean test internally and just enhancing the returned/yielded dataset status record with e.g. extra field
```
"extra_statuses": {
  "clean": [ ... records from clean  ... ]
}
```
or alike, which we could later enhance possibly with other "hints" or information (aren't we getting more info from `git annex info` which might just get here into `annex-info` ), and then leave it to renderer to just take care about rendering them (instead of running those extra checks)?

Reply:
@yarikoptic  Hm. Two things speak against that from my POV:
1. After @mih chimed in, I explicitly changed from yielding anything about that to only call `clean --info` from within the summary renderer. See https://github.com/datalad/datalad/pull/5738#pullrequestreview-682620241. I think the argument makes sense.
2. There are clean results coming from a direct invocation of `clean`. I think it's strange that - when called from within `status` - you'd have the info in an extra field of the status-result, while when calling directly you get it as `clean-info` results. That seems inconsistent with how we deal with results and such an approach would generally make it harder to act upon results via hooks - you'd need know all the forms in which it can appear. So, I think if it's something that genuinely yielded from another command it should stay that way. Otherwise I'm fine with extra fields.